core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -654,6 +654,16 @@ package object config {
       .booleanConf
       .createWithDefault(false)
 
+  private[spark] val SHUFFLE_SERVICE_REMOVE_SHUFFLE_ENABLED =
+    ConfigBuilder("spark.shuffle.service.removeShuffle")
+      .doc("Whether to use the ExternalShuffleService for deleting shuffle blocks for " +
+        "deallocated executors when the shuffle is no longer needed. Without this enabled, " +
+        "shuffle data on executors that are deallocated will remain on disk until the " +
+        "application ends.")
+      .version("3.3.0")
```

Comment:
We need to discuss about this, @mridulm .
If this feature is not urgent, I'd like to recommend to have this `master` only.

Reply:
We had to disable the external shuffle service for basically all of our large shuffling jobs until I got this working, and based on others interest and all the jira tickets that have been made over the years I'd consider this a critically missing feature. I'm fine disabling by default, especially if we try to limit the permission changes anymore, the more likely this will cause issues using the external shuffle service

Reply:
If there are concerns with stability, I am fine with pushing this to master only - in which case we can change all the versions to `3.4.0`
Thoughts @tgravescs ?

Reply:
I don't have strong opinion. I'm fine with whatever others think.  It might be best to at least turn it off by default if it goes into 3.3.